api/analyzers/java/analyzer.py

```
@@ -0,0 +1,223 @@
+from tree_sitter import Language, Parser, Node
+
+JAVA_LANGUAGE = Language(tsjava.language())
+
+import logging
+logger = logging.getLogger('code_graph')
+
+class JavaAnalyzer(AbstractAnalyzer):
+    def __init__(self) -> None:
+        self.parser = Parser(JAVA_LANGUAGE)
+
+    def process_method_declaration(self, node: Node, path: Path, source_code: str) -> Optional[Function]:
```

Ruff (0.8.2)

`Function` may be undefined, or defined from star imports

(F405)